src/main/java/com/epam/internship/carrental/car/CarServiceImpl.java

```diff
@@ -0,0 +1,234 @@
+package com.epam.internship.carrental.car;
+
+import com.epam.internship.carrental.car.enums.CarGearbox;
+import com.epam.internship.carrental.car.enums.CarType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+/**
+ * Implementation of the {@link com.epam.internship.carrental.car.CarService} interface.
+ */
+@Service
+@Qualifier("carService")
+public class CarServiceImpl implements CarService {
+
+    private static final String AUTH_HEADER = "Token ";
+    private static final String AUTH_TOKEN = AUTH_HEADER + "token";
+
+    private final CarRepository carRepository;
+
+    @Autowired
+    public CarServiceImpl(final CarRepository carRepository) {
+        this.carRepository = carRepository;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param make       the make of the car
+     * @param model      model of the car
+     * @param carType    carType of the car
+     * @param seats      number of seats of the car
+     * @param fuelUsage  fuel usage of the car
+     * @param carGearbox gearbox of the car
+     * @return ResponseEntity with Response Code 200 on success, 403 on failure.
+     */
+    @Override
+    public ResponseEntity addNewCar(final String make, final String model,
+                                    final CarType carType, final int seats,
+                                    final double fuelUsage, final CarGearbox carGearbox) {
+        if (carRepository.existByCar(make, model, carType, seats, fuelUsage, carGearbox)) {
+            return new ResponseEntity(HttpStatus.FORBIDDEN);
```

Comment:
The service layer should not know that this will be used by HTTP service, hence it should not return `ResponseEntity`. The service layer must be independent from the layer above, it should not really know anything about it.